src/Infrastructure.Dapper/Repositories/UserRepository.cs

```diff
@@ -254,6 +254,46 @@ public async Task UpdateUserKeyAndEncryptedDataAsync(
     }
 
 
+    public async Task UpdateUserKeyAndEncryptedDataV2Async(
+        User user,
+        IEnumerable<UpdateEncryptedDataForKeyRotation> updateDataActions)
+    {
+        await using var connection = new SqlConnection(ConnectionString);
+        connection.Open();
+
+        await using var transaction = connection.BeginTransaction();
+        try
+        {
+            if (user.AccountRevisionDate == null)
+            {
+                user.AccountRevisionDate = user.RevisionDate;
+            }
+
+            ProtectData(user);
+            await connection.ExecuteAsync(
+                $"[{Schema}].[{Table}_Update]",
+                user,
+                transaction: transaction,
+                commandType: CommandType.StoredProcedure);
+
+            //  Update re-encrypted data
+            foreach (var action in updateDataActions)
+            {
+                await action(connection, transaction);
+            }
+            transaction.Commit();
+        }
+        catch
+        {
+            connection.Close();
+            UnprotectData(user);
+            throw;
+        }
+        connection.Close();
```

Comment:
Do we need the `connection.Close()` here?

I thought the using statements handle disposing and closing those things in dapper?